py3status/modules/updates.py

```diff
@@ -0,0 +1,215 @@
+}
+```
+
+SAMPLE OUTPUT
+[{'full_text': 'Apk '}, {'full_text': '3', 'color': '#a9a9a9'}]
+
+14pkgs
+[{'full_text': 'Apt '}, {'full_text': '14', 'color': '#00FF00'}]
+
+29and5pkgs
+[{'full_text': 'Pacman '}, {'full_text': '29 ', 'color': '#FFFF00'},
+{'full_text': 'Auracle '}, {'full_text': '5', 'color': '#a9a9a9'}]
+
+35pkgs
+[{'full_text': 'Zypper '}, {'full_text': '34', 'color': '#ffa500'}]
+
+45pkgs
+[{'full_text': 'Xbps '}, {'full_text': '45', 'color': '#FF0000'}]
+
+no_updates
+{'full_text': 'No Updates'}
+"""
+
+STRING_INVALID = 'invalid manager'
+
+
+class Update:
+    def __init__(self, parent, name, command):
+        self.parent = parent
+        self.name = name
+        self.command = command
+
+    def get_output(self):
+        try:
+            return self.parent.py3.command_output(self.command)
+        except self.parent.py3.CommandError:
+            return None
+
+    def count_updates(self, output):
+        return len(output.splitlines())
+
+    def get_updates(self):
+        output = self.get_output()
+        if output is None:
+            return {self.name: output}
+        return {self.name: self.count_updates(output)}
+
+
+class Apt(Update):
+    def count_updates(self, output):
+        return len(output.splitlines()[1:])
+
+
+class Apk(Update):
+    def count_updates(self, output):
+        return len(output.splitlines()[1:])
+
+
+class Eopkg(Update):
+    def get_output(self):
+        output = self.parent.py3.command_output(self.command)
+        if "No packages to upgrade." in output:
+            return ""
+        return output
+
+
+class Zypper(Update):
+    def count_updates(self, output):
+        return len([x for x in output.splitlines() if x][4:])
+
+
+class Py3status:
+    """
+    """
+
+    # available configuration parameters
+    cache_timeout = 600
+    format = None
+    thresholds = [
+        (0, 'darkgray'), (10, 'degraded'), (20, 'orange'), (30, 'bad')]
+
+    class Meta:
+        deprecated = {
+            "rename_placeholder": [
+                {
+                    "placeholder": "aur",
+                    "new": "auracle",
+                    "format_strings": ["format"],
+                }
+            ]
+        }
+
+    def post_config_hook(self):
+        managers = update_managers = [
+            ("Pacman", "checkupdates"),
+            ("Auracle", "auracle sync --color=never"),
+            ("Yay", "yay --query --upgrades --aur"),
+            ("Apk", "apk version -l '<'"),
+            ("Apt", "apt list --upgradeable"),
+            ("Eopkg", "eopkg list-upgrades"),
+            ("Pkg", "pkg upgrade --dry-run --quiet"),
+            ("Xbps", "xbps-install --update --dry-run"),
+            ("Zypper", "zypper list-updates"),
+        ]
+
+        managed = getattr(self, 'managers', None)
+        if managed:
+            new_managers = []
+            for custom in managed:
+                if isinstance(custom, tuple):
```

Comment:
don't introduce multiple formats for this parameter, let `managers` only be a list of tuple and nothing else

Reply:
This debug config permits me to use `managers = ['pacman', 'yay']` without tuples (by checking against `update_managers`) to ignore auracle, cower, etc to make sure everything works okay.

You even can mix it. It's my debug config. \o/

```
updates {
    managers = [
        'pacman',
        ('Auracle', 'auracle sync'),
         'yay'
    ]
}
```

I expect most users to add module and forget it... And if they want to change placeholders, they would use `format`, not `managers`. \o/

Reply:
I understand what it permits, I just think it's a bad idea, having one way of configuring (via tuple) is more than enough and saves you a whole branch of code.